lib/graph.ml

```diff
@@ -0,0 +1,90 @@
+type color = Red | Blue | Green | Yellow
```

Comment:
I thought I would put an int. As the Mr Rabie had shown in his examples.

Reply:
It's mainly to better differentiate them from the coordinates, they're always comparable too, that's why I made this choice, after it's true that it restricts the colors to a finite number, I'd surely make the change